src/components/Launch/LaunchWorkflowForm/useLaunchWorkflowFormState.ts

```diff
@@ -213,32 +320,70 @@ export function useLaunchWorkflowFormState({
         setParsedInputs(parsedInputs);
     }, [workflow.hasLoaded, workflow.value, launchPlanData]);
 
-    // Once workflows have loaded, attempt to select the first option
+    // Once workflows have loaded, attempt to select the preferred workflow
+    // plan, or fall back to selecting the first option
     useEffect(() => {
         if (workflowSelectorOptions.length > 0 && !selectedWorkflow) {
+            if (preferredWorkflowId) {
+                const preferred = workflowSelectorOptions.find(({ data }) =>
+                    isEqual(data, preferredWorkflowId)
+                );
+                if (preferred) {
+                    setWorkflow(preferred);
+                    return;
+                }
+            }
             setWorkflow(workflowSelectorOptions[0]);
         }
     }, [workflows.value]);
 
-    // Once launch plans have been loaded, attempt to select the default
-    // launch plan
+    // Once launch plans have been loaded, attempt to keep the previously
+    // selected launch plan, followed by the preferred launch plan, the one
+    // matching the workflow name, or just the first option.
     useEffect(() => {
         if (!launchPlanSelectorOptions.length) {
             return;
         }
+
+        if (lastSelectedLaunchPlanName) {
+            const lastSelected = launchPlanSelectorOptions.find(
+                ({ name }) => name === lastSelectedLaunchPlanName
+            );
+            if (lastSelected) {
+                onSelectLaunchPlan(lastSelected);
+                return;
+            }
+        }
+
+        if (preferredLaunchPlanId) {
+            const preferred = launchPlanSelectorOptions.find(
```

Comment:
We're not worried about this being an intentional `''`, right?

Reply:
The `id` property is an object, so no. It's either defined with the expected fields, or it's falsey and we want to ignore it.